### PR TITLE
[Merged by Bors] - feat: remove triangularizability assumption from proof that Cartan subalgebras are Abelian

### PR DESCRIPTION
**Pull request by ocfnash**
---

- [x] depends on: #8572

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/Killing.lean [hunk 1]
```diff
@@ -386,7 +397,7 @@ instance isSemisimple [IsDomain R] [IsPrincipalIdealRing R] : IsSemisimple R L :
 
 instance instIsLieAbelian_of_isCartanSubalgebra
     [IsDomain R] [IsPrincipalIdealRing R] [IsArtinian R L]
-    (H : LieSubalgebra R L) [H.IsCartanSubalgebra] [LieModule.IsTriangularizable R H L] :
```
Comment (ocfnash) [inline Mathlib/Algebra/Lie/Killing.lean:389]: Dropping this typeclass assumption is the whole point of this PR.

File: Mathlib/Algebra/Lie/Killing.lean [hunk 2]
```diff
@@ -210,18 +212,27 @@ lemma lowerCentralSeries_one_inf_center_le_ker_traceForm :
...
-  replace hz : LinearMap.trace R _ (toEndomorphism R L (weightSpaceOf M χ x) z) = 0 :=
-    trace_toEndomorphism_eq_zero_of_mem_lcs R L _ (le_refl _) hz
-  let f := toEndomorphism R L (weightSpaceOf M χ x) z
-  let g := toEndomorphism R L (weightSpaceOf M χ x) x
-  have h_comm : Commute f g := commute_toEndomorphism_of_mem_center_left _ hzc x
-  have hg : _root_.IsNilpotent (g - algebraMap R _ χ) :=
-    (toEndomorphism R L M x).isNilpotent_restrict_iSup_sub_algebraMap χ
-  rw [traceForm_apply_apply, LinearMap.trace_comp_eq_mul_of_commute_of_isNilpotent χ h_comm hg, hz,
-    mul_zero]
+  rw [traceForm_apply_apply, LinearMap.zero_apply]
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Killing.lean:215]: Would you mind commenting this proof a bit?

**Review by jcommelin — APPROVED** [review]

File: Mathlib/Algebra/Lie/Killing.lean (hunk 2 above)
Comment (ocfnash) [reply Mathlib/Algebra/Lie/Killing.lean:235]: Good idea, done!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**